components/Navigation/ItemMobileNavigation.tsx

```diff
@@ -1,51 +1,47 @@
 import Link from "next/link";
 import React, { useState } from "react";
 import { Typography, VariantsTypography } from "../../ui/typography";
-import { ILinkNavigation } from "./Navigation";
 import {
   HoverLinkSC,
   LinkSC,
   SubcategorydLinkSC,
   SubNavigationMobile,
 } from "./styles";
+import { ILink, ILinkSubcategories } from "./types";
 
-export interface ILinkSubcategories {
-  name: string;
-  link: string;
-}
+export const ItemMobileNavigation = ({ link }: ILink) => {
+  const [open, isOpen] = useState(false);
 
-export const ItemMobileNavigation = ({
-  subcategories,
-  title,
-  href,
-  iconClose,
-  iconOpen,
-}: ILinkNavigation) => {
-  const [subnav, setSubnav] = useState<boolean>(false);
+  const handleMouseEnter = () => {
+    isOpen(true);
+  };
+  const handleMouseLeave = () => {
+    isOpen(false);
+  };
 
-  const showSubNav = () => setSubnav(!subnav);
   return (
     <>
-      <LinkSC onMouseEnter={subcategories && showSubNav}>
-        <Link href={href}>
+      <LinkSC
+        onMouseLeave={handleMouseLeave}
+        onMouseEnter={link.subcategories && handleMouseEnter}
+      >
+        <Link href={link.href}>
           <HoverLinkSC>
             <Typography variant={VariantsTypography.subtitle}>
-              {title}
+              {link.title}
             </Typography>
-            {subcategories && subnav
-              ? iconOpen
-              : subcategories
-              ? iconClose
-              : null}
+            {link.subcategories && link.iconOpen}
           </HoverLinkSC>
         </Link>
-        {subnav && (
-          <SubNavigationMobile>
-            {subcategories?.map((item: ILinkSubcategories, index: number) => (
-              <Link href={item.link} key={index}>
-                <SubcategorydLinkSC>{item.name}</SubcategorydLinkSC>
-              </Link>
-            ))}
+        {open && (
+          <SubNavigationMobile open={open}>
+            {link.subcategories?.map(
+              (item: ILinkSubcategories, index: number) => (
```

Comment:
types? If you are not have any types, TS will automatically understand which types will have item and index and you can cut it